stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/MessageItem.kt

```diff
@@ -784,6 +815,80 @@ internal fun DefaultMessageTextContent(
     }
 }
 
+/**
+ * Represent a swipe to reply content.
+ *
+ * @param modifier Modifier for styling.
+ * @param onReply Handler when the user swipes to reply.
+ * @param isSwipeable Handler to determine if the message is swipeable.
+ * @param swipeToReplyContent The content to show when swiping to reply.
+ * @param content The swipeable content to show when not swiping to reply.
+ */
+@Composable
+private fun SwipeToReply(
+    modifier: Modifier = Modifier,
+    onReply: () -> Unit = {},
+    isSwipeable: () -> Boolean = { true },
+    swipeToReplyContent: @Composable RowScope.() -> Unit,
+    content: @Composable () -> Unit,
+) {
+    var swipeToReplyWith by remember { mutableFloatStateOf(0f) }
+    var rowWith by remember { mutableFloatStateOf(0f) }
+    val offset = remember { Animatable(initialValue = 0f) }
+    val scope = rememberCoroutineScope()
+
+    Box(
+        modifier = modifier
+            .fillMaxWidth(),
+    ) {
+        Row(
+            modifier = Modifier
+                .align(Alignment.CenterStart)
+                .onSizeChanged { swipeToReplyWith = it.width.toFloat() }
+                .offset {
+                    val roundToInt = swipeToReplyWith.roundToInt()
+                    IntOffset(
+                        (offset.value.roundToInt() - roundToInt)
+                            .coerceIn(-roundToInt, roundToInt),
+                        0,
+                    )
+                },
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            swipeToReplyContent()
+        }
+        Row(
+            modifier = modifier
+                .fillMaxWidth()
+                .onSizeChanged { rowWith = it.width.toFloat() }
+                .offset { IntOffset(offset.value.roundToInt(), 0) }
+                .pointerInput(swipeToReplyWith) {
+                    if (isSwipeable()) {
+                        detectHorizontalDragGestures(
+                            onHorizontalDrag = { _, dragAmount ->
+                                scope.launch {
+                                    val newOffset = (offset.value + dragAmount)
+                                        .coerceIn(0f, maxOf((rowWith / 2), swipeToReplyWith))
+                                    offset.snapTo(newOffset)
+                                }
+                            },
+                            onDragEnd = {
+                                scope.launch {
+                                    onReply
+                                        .takeIf { offset.value >= swipeToReplyWith }
+                                        ?.invoke()
```

Comment:
It would be nice if we could add haptic feedback to provide a nice UX.
Something like this:
```kotlin
val shouldReply = offset.value >= swipeToReplyWith
if (shouldReply) {
    view.performHapticFeedback(HapticFeedbackConstantsCompat.CONFIRM)
    onReply()
}
```
`view` could be retrieved from `LocalView.current`

Compose also has haptic feedback API, but it's too generic and limited to `LongPress` and `TextHandleMove`